internal/webhook/external/quay.go

```diff
@@ -0,0 +1,151 @@
+package external
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	xhttp "github.com/akuity/kargo/internal/http"
+	"github.com/akuity/kargo/internal/io"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+const (
+	quaySecretDataKey       = "secret"
+	quay                    = "quay"
+	quayWebhookBodyMaxBytes = 2 << 20 // 2MB
+)
+
+func init() {
+	registry.register(
+		quay,
+		webhookReceiverRegistration{
+			predicate: func(cfg kargoapi.WebhookReceiverConfig) bool {
+				return cfg.Quay != nil
+			},
+			factory: newQuayWebhookReceiver,
+		},
+	)
+}
+
+// quayWebhookReceiver is an implementation of WebhookReceiver that handles
+// inbound webhooks from Quay.
+type quayWebhookReceiver struct {
+	*baseWebhookReceiver
+}
+
+// newQuayWebhookReceiver returns a new instance of quayWebhookReceiver.
+func newQuayWebhookReceiver(
+	c client.Client,
+	project string,
+	cfg kargoapi.WebhookReceiverConfig,
+) WebhookReceiver {
+	return &quayWebhookReceiver{
+		baseWebhookReceiver: &baseWebhookReceiver{
+			client:     c,
+			project:    project,
+			secretName: cfg.Quay.SecretRef.Name,
+		},
+	}
+}
+
+// GetDetails implements WebhookReceiver.
+func (q *quayWebhookReceiver) getReceiverType() string {
+	return quay
+}
+
+// getSecretValues implements WebhookReceiver.
+func (q *quayWebhookReceiver) getSecretValues(
+	secretData map[string][]byte,
+) ([]string, error) {
+	secretValue, ok := secretData[quaySecretDataKey]
+	if !ok {
+		return nil,
+			errors.New("Secret data is not valid for a Quay WebhookReceiver")
+	}
+	return []string{string(secretValue)}, nil
+}
```

Comment:
We don't appear to be doing anything with `secretValue`. Do we need to? Or do we not need this info here?

I've not read up on Quay webhooks, but either this plays a role in authenticating the request somehow, or if it doesn't (e.g. case where the unguessable URL is is own implicit authentication mechanism), then we don't need to do anything with it. Not sure which of those it is.

Reply:
Ok... Just drop this whole stanza then.